src/autoscaler/operator/scalingenginedb.go

```diff
@@ -21,17 +23,24 @@ func NewScalingEngineDbPruner(scalingEngineDb db.ScalingEngineDB, cutoffDuration
 		scalingEngineDb: scalingEngineDb,
 		cutoffDuration:  cutoffDuration,
 		clock:           clock,
-		logger:          logger,
+		logger:          logger.Session("scaling_engine_db_pruner"),
 	}
 }
 
-func (sdp ScalingEngineDbPruner) Operate() {
-	sdp.logger.Debug("Pruning  scaling histories")
-
+func (sdp ScalingEngineDbPruner) Operate(ctx context.Context) {
 	timestamp := sdp.clock.Now().Add(-sdp.cutoffDuration).UnixNano()
-	err := sdp.scalingEngineDb.PruneScalingHistories(timestamp)
+
+	logger := sdp.logger.Session("pruning-instance-metrics", helpers.AddTraceID(ctx, lager.Data{"cutoff-time": timestamp}))
```

Comment:
Just out of curiosity: logging the trace id is done to correlate log entries and otel traces? Is this also the way to go according to OpenTelemetry and their logging approach?

Reply:
Yes, to the first question.
No to second question, as it as of now unknown how the OpenTelemetry approach will look like, see https://github.com/open-telemetry/opentelemetry-go?tab=readme-ov-file#project-status